test/data_structures.jl

```diff
@@ -49,3 +49,201 @@ end
     @test cache.priority == ["c", "a", "d"]
     @test length(cache.priority) == length(keys(cache.cache)) == cache.max_size
 end
+
+@testset "get with default value" begin
+    cache = DataStructures.LRUCache{String, Int}(max_size = 3)
+
+    # Test cache hits (key in dict)
+    get!(cache, "a", 1)
+    get!(cache, "b", 2)
+    @test get(cache, "a", 3) == 1
+    @test cache.priority == ["b", "a"]
+
+    # check cache miss (key not in dict)
+    @test get(cache, "c", 3) == 3
+    @test get(cache, "c", 2) == 2
+    @test cache.priority == ["b", "a"]
+end
+
+@testset "get with default callable" begin
+    cache = DataStructures.LRUCache{String, Int}(max_size = 3)
+
+    # Test cache hits (key in dict)
+    get!(cache, "a", 1)
+    get!(cache, "b", 2)
+    @test get(() -> 3, cache, "a") == 1
+    @test cache.priority == ["b", "a"]
+
+    # # check cache miss (key not in dict)
+    @test get(() -> 3, cache, "c") == 3
+    @test get(() -> 2, cache, "c") == 2
+    @test cache.priority == ["b", "a"]
+end
+
+@testset "haskey LRU" begin
+    cache = DataStructures.LRUCache{String, Int}(max_size = 3)
+
+    # Test haskey if key exists
+    get!(cache, "a", 1)
+    get!(cache, "b", 2)
+    @test haskey(cache, "a") == true
```

Comment:
```suggestion
    @test haskey(cache, "a")
```

`haskey` already returns a boolean, you don't need to check that is equal to true.